datastore/src/main/java/org/jboss/pnc/datastore/DefaultDatastore.java

```diff
@@ -246,7 +246,8 @@ private void fetchOrSaveRequiredTargetRepositories(
         for (Artifact artifact : artifacts) {
             TargetRepository targetRepository = artifact.getTargetRepository();
             logger.trace("Adding repository for artifact: {}.", artifact.toString());
-            if (!storedTargetRepositories.containsKey(targetRepository.getIdentifierPath())) {
+            if (!storedTargetRepositories.containsKey(targetRepository.getIdentifierPath())
+                    && !requiredTargetRepositories.containsKey(targetRepository.getIdentifierPath())) {
```

Comment:
Why was this an issue? `requiredTargetRepositories` is a map so it doesn't contain the same key twice